Realm/ObjectStore/shared_realm.cpp

```diff
@@ -518,5 +518,50 @@ util::Optional<int> Realm::file_format_upgraded_from_version() const
     return util::Optional<int>();
 }
 
+struct Realm::HandoverPackage {
+    SharedGroup::VersionID version;
+    std::vector<AnyHandover> objects;
+
+    bool is_awaiting_import() {
+        return version != SharedGroup::VersionID();
+    }
+
+    ~HandoverPackage() {
+        if (is_awaiting_import()) {
+            REALM_TERMINATE("Handover package not imported, leaking pinned version");
+        }
+    }
+};
+
+std::shared_ptr<Realm::HandoverPackage> Realm::package_for_handover(std::vector<AnyThreadConfined> objects_to_hand_over) {
+    std::shared_ptr<Realm::HandoverPackage> handover = std::make_shared<HandoverPackage>();
```

Comment:
The type of the value is explicitly spelled-out on the RHS, so this is one of the clearcut places for using `auto`.